src/aap_eda/core/models/eda_credential.py

```diff
@@ -0,0 +1,39 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from django.db import models
+
+__all__ = ("EdaCredential",)
+
+
+class EdaCredential(models.Model):
+    class Meta:
+        db_table = "core_eda_credential"
+        constraints = [
+            models.CheckConstraint(
+                check=~models.Q(name=""),
+                name="ck_empty_eda_credential_name",
+            )
+        ]
+
+    name = models.TextField(null=False, unique=True)
+    description = models.TextField(default="", blank=True, null=False)
+    inputs = models.JSONField(default=dict)
```

Comment:
shouldn't this field be EncryptedTextField? Its plain value is the serialized inputs.

Reply:
> shouldn't this field be EncryptedTextField? Its plain value is the serialized inputs.

Is it encrypted by our db secret? or encrypted by the system vault?

Reply:
changed to EncryptedTextField